ledger/README.md

```diff
@@ -7,3 +7,59 @@ This package provides two ledger implementations:
 
 - **Partial Ledger** implements the ledger functionality for a limited subset of keys. Partial ledgers are designed to be constructed and verified by a collection of proofs from a complete ledger. The partial ledger uses a partial binary Merkle trie which holds intermediate hash value for the pruned branched and prevents updates to keys that were not part of proofs.
+
+## Definitions
+In this section we provide an overview of some of the concepts. Hence it is highly recommended to checkout [this doc](https://github.com/onflow/flow-go/blob/master/ledger/complete/mtrie/README.md) for the formal and technical definitions in more details.
+
+### binary Merkle tree
+
+![binary Merkle tree image](/ledger/docs/binary_merkle_tree.png?raw=true "binary Merkle tree" )
+
+In this context a *binary Merkle tree* is defined as [perfect binary tree](https://xlinux.nist.gov/dads/HTML/perfectBinaryTree.html) with a specific height including three type of nodes:
+
+- leaf nodes: holds a payload (data), a path (where the node is located), and a hash value (hash of path and payload content)
+
+- empty leaf nodes: doesn't hold any data and only stores a path, and a default hash value based on the height of tree
+
+- interim nodes: holds a hash value which is defined as hash of hash value of left and right children.
+
+![node types image](/ledger/docs/node_types.png)
+
+A *path* is a unique address of a node storing a payload. Paths are derived from the content of payloads (see common/pathfinder).
+
+![paths image](/ledger/docs/paths.png?raw=true "paths")
+
+#### Operations
+
+**Get**: Fetching a payload from the binary Merkle tree is by traversing the tree based on path bits. (0: left branch, 1: right branch)
+
+**Update**: Updates to the tree starts with traversing the tree to the leaf node, updating payload, hash value of that node and hash value of all the ancestor nodes (nodes on higher level connected to this node).
+
+![update image](/ledger/docs/tree_update.gif?raw=true "update")
+
+**Prove**: A binary Merkle tree can provide an inclusion proof for any given payload. A Merkle proof in this context includes all the information needed to walk through a tree branch from an specific leaf node (key) up to the root of the tree (yellow node hash values are needed for inclusion proof for the green node).
+
+![proof image](/ledger/docs/proof.png?raw=true "proof")
+
+### binary Merkle trie
+A **binary Merkle trie** in this context is defined as a compact version of binary Merkle tree, providing exact same functionality but doesn't explicitly store empty nodes. Formally, a node is empty: 
+* the node is an empty leaf node (see definition above)
```

Comment:
Once we remove the empty leaf node from above, we would need to define it here:

```suggestion
* the node is an empty leaf node: it doesn't hold any data and only stores a path. Its hash value is defined as a default hash based on the height of tree.
```

Reply:
Applied at 2ba9d6c